nixos/modules/system/activation/switch-to-configuration.pl

```diff
@@ -226,10 +239,20 @@ sub unrecord_unit {
 sub compare_units {
     my ($old_unit, $new_unit) = @_;
     my $ret = 0;
+    # Keys to ignore in the [Unit] section
+    my %unit_section_ignores = map { $_ => 1 } qw(
+        X-Reload-Triggers
+        Description Documentation
+        OnFailure OnSuccess OnFailureJobMode
+        IgnoreOnIsolate StopWhenUnneeded
+        RefuseManualStart RefuseManualStop
+        AllowIsolate CollectMode
+        SourcePath
+    );
 
     my $comp_array = sub {
       my ($a, $b) = @_;
-      return join("\0", @{$a}) eq join("\0", @{$b});
+      return join("\0", @{$a}) eq join "\0", @{$b};
```

Comment:
Why drop the parens around the last join?

Reply:
That is a very good question. It's because of https://metacpan.org/pod/Perl::Critic::Policy::CodeLayout::ProhibitParensWithBuiltins

Now since there is no requirement to be 100% critic-compatible: Do you think we should just ignore this one rule in general? I somewhat understand their argument but when nesting builtin functions (like `join(split(…`) the code quickly becomes less readable imo
If you think we should ignore the rule I will rebase out that change

Reply:
Imho; the `perlcritic` suggestion doesn't make sense in this case, as dropping the parens for the last join() call makes it inconsistent with the first, and makes them appear different.

Reply:
I agree with your point.
So much so that I didn't just rebase the change in but rather added another commit that adds parenthesis to **all** calls in the script.

Reply:
:+1: Not my preference, but think [it's fine](https://en.wikipedia.org/wiki/There%27s_more_than_one_way_to_do_it) as long as it is consistent.